tests/kubernetes-e2e.rs

```diff
@@ -0,0 +1,732 @@
+//! This test is optimized for very quick rebuilds as it doesn't use anything
+//! from the `vector` crate, and thus doesn't waste time in a tremendously long
+//! link step.
+
+use futures::{SinkExt, StreamExt};
+use k8s_openapi::{
+    api::core::v1::{Container, Pod, PodSpec},
+    apimachinery::pkg::apis::meta::v1::ObjectMeta,
+};
+use k8s_test_framework::{
+    lock, test_pod, wait_for_resource::WaitFor, Framework, Interface, Reader,
+};
+use std::collections::HashSet;
+
+const VECTOR_CONFIG: &str = r#"
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: vector-config
+data:
+  vector.toml: |
```

Comment:
I really think we should teach Vector to handle YAML so this can look slick!

Reply:
Good idea! This could make Helm config much more flexible too.